starknet-foundry/crates/forge/src/scarb.rs

```diff
@@ -140,7 +140,14 @@ pub fn dependencies_for_package(
 
     let target_name = compilation_unit.target.name.clone();
 
-    Ok((base_path, dependencies, target_name))
+    let corelib = compilation_unit
+        .components
+        .iter()
+        .find(|du| du.source_path.to_string().contains("core/src"))
+        .context("corelib could not be found")?;
+    let corelib_path = Utf8PathBuf::from(corelib.source_root());
+
+    Ok((base_path, corelib_path, dependencies, target_name))
```

Comment:
This function returns a corelib, but for now the corelib bundled with scarb is not compatible with the compiler version we use (see comments in `Cargo.toml`). It is prepared to be used when 2.1.0 is release - we can then bump Cairo to 2.1.0 and use the corelib provided by appropriate Scarb version